doc/stdenv/stdenv.chapter.md

```diff
@@ -95,6 +95,27 @@ installPhase() {
 genericBuild
 ```
 
+### Building a `stdenv` package in `nix-shell` {#sec-building-stdenv-package-in-nix-shell}
+
+To build a `stdenv` package in a `nix-shell`, use
+
+```bash
+nix-shell '<nixpkgs>' -A some_package
+eval ${unpackPhase:-unpackPhase}
+cd $sourceRoot
+eval ${patchPhase:-patchPhase}
+eval ${configurePhase:-configurePhase}
+eval ${buildPhase:-buildPhase}
+```
+
+To modify a phase, first print it with
```

Comment:
```suggestion
To modify a [phase], first print it with
```

Please link to the phases overview from here. We can expect people to read such things in isolation before even knowing what phases are.